File: app/src/main/java/com/patrickiv/demo/enteranimationdemo/MainActivity.java
package com.patrickiv.demo.enteranimationdemo;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;

import com.patrickiv.demo.enteranimationdemo.fragment.GridDemoFragment;
import com.patrickiv.demo.enteranimationdemo.fragment.ListDemoFragment;

/**
 * Created by Patrick Ivarsson on 7/17/17.
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private ViewGroup mButtonContainer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mButtonContainer = findViewById(R.id.button_container);
        findViewById(R.id.btn_list_fragment).setOnClickListener(this);
        findViewById(R.id.btn_grid_fragment).setOnClickListener(this);

        if (savedInstanceState != null) {
            final int count = getSupportFragmentManager().getBackStackEntryCount();
            mButtonContainer.setAlpha(count > 0 ? 0f : 1f);
        }
    }

    @Override
    public void onBackPressed() {
        if (getSupportFragmentManager().getBackStackEntryCount() == 1) {
            mButtonContainer.animate().alpha(1f).start();
        }
        super.onBackPressed();
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_list_fragment:
                showFragment(new ListDemoFragment());
                break;
            case R.id.btn_grid_fragment:
                showFragment(new GridDemoFragment());
                break;
            default:
                break;
        }
    }

    private void showFragment(Fragment fragment) {
        getSupportFragmentManager().beginTransaction()
                .add(R.id.container, fragment)
                .addToBackStack(null)
                .commit();
        mButtonContainer.animate().alpha(0f).start();
    }

}

File: app/src/main/java/com/patrickiv/demo/enteranimationdemo/MainActivity.kt
package com.patrickiv.demo.enteranimationdemo

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.patrickiv.demo.enteranimationdemo.fragment.GridDemoFragment
import com.patrickiv.demo.enteranimationdemo.fragment.ListDemoFragment
import kotlinx.android.synthetic.main.activity_main.buttonContainer
import kotlinx.android.synthetic.main.activity_main.gridSample
import kotlinx.android.synthetic.main.activity_main.headerSample
import kotlinx.android.synthetic.main.activity_main.listSample

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        listSample.setOnClickListener(this)
        headerSample.setOnClickListener(this)
        gridSample.setOnClickListener(this)

        supportFragmentManager.addOnBackStackChangedListener {
            buttonContainer.alpha = if (supportFragmentManager.backStackEntryCount > 0) 0f else 1f
        }
    }

    override fun onClick(view: View) {
        val fragment = when (view.id) {
            R.id.listSample -> ListDemoFragment.instance(withHeader = false)
            R.id.headerSample -> ListDemoFragment.instance(withHeader = true)
            else -> GridDemoFragment()
        }
        supportFragmentManager.beginTransaction()
            .replace(R.id.container, fragment)
            .addToBackStack(null)
            .commit()
    }
}